proposals/nnnn-access-level-on-imports.md

```diff
@@ -0,0 +1,321 @@
+
+In Swift 6 mode an import is internal by default.
+This will align the behavior of imports with declarations where the implicit access-level is internal.
+It should help limit unintentional dependency creep as marking a dependency public will require an intentional extra keyword.
+
+The Swift 6 change will likely break source compatibility for libraries.
+A migration tool could automatically insert the `public` keyword where required.
+Where the tool is unavailable, a simple script can insert a `public` keyword in front of all imports to preserve the Swift 5 behavior.
+
+The upcoming feature flag `InternalImports` will enable the Swift 6 behavior even when using Swift 5.
+
+### Relation with other attributes on imports
+
+The `@_exported` attribute is a step above a `public` import
+as clients see the imported module declarations is if they were part of the local module.
+With this proposal, `@_exported` is accepted only on public import statements,
+both with the keyword or the default public visibility in Swift 6 mode.
+
+The `@testable` attribute allows the local module to reference the internal declarations of the imported module.
+The current design even allows to use an imported internal or package type in a public declaration.
+The access-level behavior applies in the same way as a normal import,
+all imported declarations have as upper-bound the access-level on the import statement.
+In the case of a `@testable` import, even the imported internal declarations are affected by the bound.
+
+## Source compatibility
+
+To preserve source compatibility, imports are public by default in Swift 5.
+This will preserve the current behavior of imports in Swift 5.
+As discussed previously, the Swift 6 behavior changes the default value and will require code changes.
+
+## ABI compatibility
+
+This proposal doesn't affect ABI compatibility,
+it is a compile time change enforced by type-checking.
+
+## Implications on adoption
+
+Adopting or reverting the adoption of this feature should not affect clients if used with care.
+
+In the case of adoption in a non-resilient module, the change is in type-checking of the module source files only.
+In this case changing the access-level of different dependencies won't affect clients.
+
+For adoption in a resilient module,
+marking an existing import as less than public will affect how clients build.
+The compiler can build the clients by loading fewer transitive dependencies.
+In theory, this shouldn't affect the clients but it may still lead to different compilation behaviors.
+
+In theory, these transitive dependencies couldn't be used by the clients,
+so hiding them doesn't affect the clients.
+In practice, there are leaks allowing use of extension members from transitive dependencies.
+Adopting this feature may skip loading transitive dependencies and prevent those leaks,
+it can break source compatibility in code relying of those behaviors.
+
+### As a replacement to `@_implementationOnly`
```

Comment:
It was there from the pitch and because many adopters will be replacing `@_implementationOnly` with this feature. But I moved it, it fits well with the other attributes.